Modules/_collectionsmodule.c

```diff
@@ -1041,16 +1041,25 @@ deque_index(dequeobject *deque, PyObject *const *args, Py_ssize_t nargs)
         start = stop;
     assert(0 <= start && start <= stop && stop <= Py_SIZE(deque));
 
-    /* XXX Replace this loop with faster code from deque_item() */
-    for (i=0 ; i<start ; i++) {
-        index++;
-        if (index == BLOCKLEN) {
-            b = b->rightlink;
-            index = 0;
+    if (start > 0) {
+        i = start + deque->leftindex;
+        n = (Py_ssize_t)((size_t) i / BLOCKLEN);
+        index = (Py_ssize_t)((size_t) i % BLOCKLEN);
+
+        if (index < (Py_SIZE(deque) >> 1)) {
+            while (--n >= 0)
```

Comment:
`braces are required everywhere, even where C permits them to be omitted ...  All new C code requires braces.` ([PEP 7](https://www.python.org/dev/peps/pep-0007/))